### DISCUSSION TITLE
[Feature Request] CUDA hardware acceleration integration

### DISCUSSION THREAD

**Discussion by z4rathustr4** [simulationcraft/simc#6660] [Ideas]
Hi there, this is my first post and I don't know neither if it's been proposed by anyone nor if it's possible with the current codebase, but here it goes:

- Is it possible to adapt the core codebase to enable CUDA hardware acceleration? 
I can imagine it would be a terrible nightmare to update all class modules and can't imagine the complexity of modifying the core for this purpose, but it has good potential when running **very** large simulations (e. g. Dungeon Slice simulations, long APLs, multiple stat scaling, also the new talent trees, which, for example Mage class tree has at a first glance a cyclomatic complexity of 15, but going into @seriallos Talent Possibilities excel, the raw count of iterations ascend to 39,483,984... not to mention DK's class nodes 😣)

Some resources: 
 - [Nvidia Developer Homepage](https://developer.nvidia.com/how-to-cuda-c-cpp)
 - [Nvidia Dev Forums - Accelerated Computing - GPU Accelerated Libraries Category](https://forums.developer.nvidia.com/c/accelerated-computing/gpu-accelerated-libraries/12)
- [Talent Possibility Excel](https://docs.google.com/spreadsheets/d/1lxTbkil8mgAhnBB7TLWK7gdcmhd-xJ5UR9obWys3MKQ/edit?usp=sharing) (credit to @seriallos obviously)

I hope it inspires the devs to add a new layer of possibilities to this wonderful project!

**Comment by scamille** [comment]
Please have a look at some of the discussion in https://github.com/simulationcraft/simc/issues/2119 . I don't think the assessment there has changed much - simc isn't really well suited for the parallelization that GPUs offer.

**Reply by z4rathustr4** [reply]
Thanks for the swift answer, I did not see that issue, I'll look at the implementation example linked there and look if I can do something better with it, just for fun 😄 .

Happy weekend.
